Can't build with extension 1.8. Error: "CMake Error: Could not create named generator =Ninja" (VSC-1450)

### OS

Windows

### Operating System version

Windows 10

### Visual Studio Code version

1.9.1

### ESP-IDF version

4.3.7

### Python version

3.8.7

### Doctor command output

---------------------------------------------- ESP-IDF Extension for Visual Studio Code report ---------------------------------------------
OS win32 x64 10.0.19045 
System environment variable IDF_PYTHON_ENV_PATH 
 undefined 
System environment variable PATH 
 C:\Program Files\Microsoft SDKs\Azure\CLI2\wbin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Git\cmd;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\dotnet\;C:<HOMEPATH>\AppData\Local\particle\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Git\cmd;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\dotnet\;C:<HOMEPATH>\AppData\Local\Microsoft\WindowsApps;;C:<HOMEPATH>\AppData\Local\Programs\Microsoft VS Code\bin;C:<HOMEPATH>\.dotnet\tools 
System environment variable PYTHON 
 undefined 
Visual Studio Code version 1.91.1 
Visual Studio Code language en 
Visual Studio Code shell C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe 
ESP-IDF Extension version 1.8.0 
Workspace folder c:\repos\espfw\controller-firmware 
---------------------------------------------------- Extension configuration settings ------------------------------------------------------
ESP-ADF Path (idf.espAdfPath) ${env:ADF_PATH}
ESP-IDF Path (idf.espIdfPath) C:<HOMEPATH>\esp\esp-idf
ESP-MDF Path (idf.espMdfPath) ${env:MDF_PATH}
ESP-Matter Path (idf.espMatterPath) ${env:ESP_MATTER_PATH}
ESP-HomeKit-SDK Path (idf.espHomeKitSdkPath) ${env:HOMEKIT_PATH}
Custom extra paths (idf.customExtraPaths) C:<HOMEPATH>\.espressif\tools\xtensa-esp32-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32-elf\bin;C:<HOMEPATH>\.espressif\tools\xtensa-esp32s2-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32s2-elf\bin;C:<HOMEPATH>\.espressif\tools\xtensa-esp32s3-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32s3-elf\bin;C:<HOMEPATH>\.espressif\tools\riscv32-esp-elf\esp-2021r2-patch3-8.4.0\riscv32-esp-elf\bin;C:<HOMEPATH>\.espressif\tools\esp32ulp-elf\2.28.51-esp-20191205\esp32ulp-elf-binutils\bin;C:<HOMEPATH>\.espressif\tools\esp32s2ulp-elf\2.28.51-esp-20191205\esp32s2ulp-elf-binutils\bin;C:<HOMEPATH>\.espressif\tools\cmake\3.16.4\bin;C:<HOMEPATH>\.espressif\tools\openocd-esp32\v0.11.0-esp32-20220706\openocd-esp32\bin;C:<HOMEPATH>\.espressif\tools\ninja\1.10.2;C:<HOMEPATH>\.espressif\tools\idf-exe\1.0.1;C:<HOMEPATH>\.espressif\tools\ccache\3.7;C:<HOMEPATH>\.espressif\tools\dfu-util\0.9\dfu-util-0.9-win64
Custom extra vars (idf.customExtraVars)
    OPENOCD_SCRIPTS: C:<HOMEPATH>\.espressif\tools\openocd-esp32\v0.11.0-esp32-20220706/openocd-esp32/share/openocd/scripts
    IDF_CCACHE_ENABLE: 1
Virtual env Python Path (idf.pythonBinPath) C:<HOMEPATH>\.espressif\python_env\idf4.3_py3.8_env\Scripts\python.exe
Serial port (idf.port) COM6
OpenOCD Configs (idf.openOcdConfigs) interface/ftdi/esp32_devkitj_v1.cfg,target/esp32.cfg
ESP-IDF Tools Path (idf.toolsPath) C:<HOMEPATH>\.espressif
Git Path (idf.gitPath) \Git\cmd\git.exe
Notification Mode (idf.notificationMode) All
-------------------------------------------------------- Configurations access -------------------------------------------------------------
Access to ESP-ADF Path (idf.espAdfPath) false
Access to ESP-IDF Path (idf.espIdfPath) true
Access to ESP-MDF Path (idf.espMdfPath) false
Access to ESP-Matter Path (idf.espMatterPath) false
Access to ESP-HomeKit Path (idf.espHomeKitSdkPath) false
Access to ESP-IDF Custom extra paths
Access to C:<HOMEPATH>\.espressif\tools\xtensa-esp32-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32-elf\bin: true
Access to C:<HOMEPATH>\.espressif\tools\xtensa-esp32s2-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32s2-elf\bin: true
Access to C:<HOMEPATH>\.espressif\tools\xtensa-esp32s3-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32s3-elf\bin: true
Access to C:<HOMEPATH>\.espressif\tools\riscv32-esp-elf\esp-2021r2-patch3-8.4.0\riscv32-esp-elf\bin: true
Access to C:<HOMEPATH>\.espressif\tools\esp32ulp-elf\2.28.51-esp-20191205\esp32ulp-elf-binutils\bin: true
Access to C:<HOMEPATH>\.espressif\tools\esp32s2ulp-elf\2.28.51-esp-20191205\esp32s2ulp-elf-binutils\bin: true
Access to C:<HOMEPATH>\.espressif\tools\cmake\3.16.4\bin: true
Access to C:<HOMEPATH>\.espressif\tools\openocd-esp32\v0.11.0-esp32-20220706\openocd-esp32\bin: true
Access to C:<HOMEPATH>\.espressif\tools\ninja\1.10.2: true
Access to C:<HOMEPATH>\.espressif\tools\idf-exe\1.0.1: true
Access to C:<HOMEPATH>\.espressif\tools\ccache\3.7: true
Access to C:<HOMEPATH>\.espressif\tools\dfu-util\0.9\dfu-util-0.9-win64: true
Access to Virtual env Python Path (idf.pythonBinPath) true
Access to CMake in environment PATH undefined
Access to Ninja in environment PATH undefined
Access to ESP-IDF Tools Path (idf.toolsPath) true
-------------------------------------------------------- Configurations has spaces -------------------------------------------------------------
Spaces in system environment Path true
Spaces in ESP-ADF Path (idf.espAdfPath) false
Spaces in ESP-IDF Path (idf.espIdfPath) false
Spaces in ESP-MDF Path (idf.espMdfPath) false
Spaces in ESP-Matter Path (idf.espMatterPath) false
Spaces in ESP-HomeKit-SDK Path (idf.espHomeKitSdkPath) false
Spaces in ESP-IDF Custom extra paths
Spaces in C:<HOMEPATH>\.espressif\tools\xtensa-esp32-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32-elf\bin: false
Spaces in C:<HOMEPATH>\.espressif\tools\xtensa-esp32s2-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32s2-elf\bin: false
Spaces in C:<HOMEPATH>\.espressif\tools\xtensa-esp32s3-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32s3-elf\bin: false
Spaces in C:<HOMEPATH>\.espressif\tools\riscv32-esp-elf\esp-2021r2-patch3-8.4.0\riscv32-esp-elf\bin: false
Spaces in C:<HOMEPATH>\.espressif\tools\esp32ulp-elf\2.28.51-esp-20191205\esp32ulp-elf-binutils\bin: false
Spaces in C:<HOMEPATH>\.espressif\tools\esp32s2ulp-elf\2.28.51-esp-20191205\esp32s2ulp-elf-binutils\bin: false
Spaces in C:<HOMEPATH>\.espressif\tools\cmake\3.16.4\bin: false
Spaces in C:<HOMEPATH>\.espressif\tools\openocd-esp32\v0.11.0-esp32-20220706\openocd-esp32\bin: false
Spaces in C:<HOMEPATH>\.espressif\tools\ninja\1.10.2: false
Spaces in C:<HOMEPATH>\.espressif\tools\idf-exe\1.0.1: false
Spaces in C:<HOMEPATH>\.espressif\tools\ccache\3.7: false
Spaces in C:<HOMEPATH>\.espressif\tools\dfu-util\0.9\dfu-util-0.9-win64: false
Spaces in Virtual env Python Path (idf.pythonBinPath) false
Spaces in ESP-IDF Tools Path (idf.toolsPath) false
----------------------------------------------------------- Executables Versions -----------------------------------------------------------
Git version undefined
ESP-IDF version undefined
Python version undefined
Python's pip version undefined
-------------------------------------------------- Project configuration settings ----------------------------------------------------------
-------------------------------------------------- Python packages in idf.pythonBinPath ----------------------------------------------------
---------------------------------------------------- Check ESP-IDF python requirements.txt -------------------------------------------------
Check ESP-IDF Python packages undefined
---------------------------------------------------- Check ESP-IDF debug adapter requirements.txt ------------------------------------------
Check Debug AdapterPython packages undefined
----------------------------------------------------------- Latest error -----------------------------------------------------------------
{
    "errno": -4058,
    "code": "ENOENT",
    "syscall": "spawn \\Git\\cmd\\git.exe",
    "path": "\\Git\\cmd\\git.exe",
    "spawnargs": [
        "--version"
    ],
    "cmd": "\\Git\\cmd\\git.exe --version",
    "vslsStack": [
        {},
        {},
        {}
    ]
}
--------------------------------------------------------------------------------------------------------------------------------------------
----------------------------------------------------------- Logfile -----------------------------------------------------------------
{"message":"Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n","stack":"Error: Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n\n\tat \\dist\\extension.js:2:1464939\n\tat ChildProcess.exithandler (node:child_process:415:7)\n\tat ChildProcess.emit (node:events:514:28)\n\tat maybeClose (node:internal/child_process:1105:16)\n\tat Socket.<anonymous> (node:internal/child_process:457:11)\n\tat Socket.emit (node:events:514:28)\n\tat Pipe.<anonymous> (node:net:337:12)","level":"error","timestamp":"2024-07-23T14:28:42.078Z"}
{"message":"Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n","stack":"Error: Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n\n\tat \\dist\\extension.js:2:1464939\n\tat ChildProcess.exithandler (node:child_process:415:7)\n\tat ChildProcess.emit (node:events:514:28)\n\tat maybeClose (node:internal/child_process:1105:16)\n\tat Socket.<anonymous> (node:internal/child_process:457:11)\n\tat Socket.emit (node:events:514:28)\n\tat Pipe.<anonymous> (node:net:337:12)","level":"error","timestamp":"2024-07-23T14:28:43.259Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:163:20722)\n\tat l.y (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:82:664)\n\tat l.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:82:881)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:156:57296)","level":"error","timestamp":"2024-07-23T14:57:29.923Z"}
{"message":"Cannot access filePath: c:\\repos\\espfw\\controller-firmware\\build\\CMakeCache.txt","stack":"Error: ENOENT: no such file or directory, access 'c:\\repos\\espfw\\controller-firmware\\build\\CMakeCache.txt'\n\tat Object.access (node:internal/fs/sync:42:11)\n\tat Object.accessSync (node:fs:248:10)\n\tat t.accessSync (node:electron/js2c/node_init:2:8888)\n\tat Se (\\dist\\extension.js:2:1461757)\n\tat \\dist\\extension.js:2:1855029","level":"error","timestamp":"2024-07-23T14:57:37.663Z"}
{"user":true,"level":"warn","message":"There is no CMakeCache.txt. Please try to delete the build directory manually.","timestamp":"2024-07-23T14:57:37.663Z"}
{"message":"Cannot access filePath: c:\\repos\\espfw\\controller-firmware\\build\\CMakeCache.txt","stack":"Error: ENOENT: no such file or directory, access 'c:\\repos\\espfw\\controller-firmware\\build\\CMakeCache.txt'\n\tat Object.access (node:internal/fs/sync:42:11)\n\tat Object.accessSync (node:fs:248:10)\n\tat t.accessSync (node:electron/js2c/node_init:2:8888)\n\tat Se (\\dist\\extension.js:2:1461757)\n\tat \\dist\\extension.js:2:1855029","level":"error","timestamp":"2024-07-23T14:57:38.842Z"}
{"user":true,"level":"warn","message":"There is no CMakeCache.txt. Please try to delete the build directory manually.","timestamp":"2024-07-23T14:57:38.842Z"}
{"message":"Cannot access filePath: c:\\repos\\espfw\\controller-firmware\\build\\CMakeCache.txt","stack":"Error: ENOENT: no such file or directory, access 'c:\\repos\\espfw\\controller-firmware\\build\\CMakeCache.txt'\n\tat Object.access (node:internal/fs/sync:42:11)\n\tat Object.accessSync (node:fs:248:10)\n\tat t.accessSync (node:electron/js2c/node_init:2:8888)\n\tat Se (\\dist\\extension.js:2:1461757)\n\tat \\dist\\extension.js:2:1855029","level":"error","timestamp":"2024-07-23T14:57:39.605Z"}
{"user":true,"level":"warn","message":"There is no CMakeCache.txt. Please try to delete the build directory manually.","timestamp":"2024-07-23T14:57:39.605Z"}
{"message":"Cannot access filePath: c:\\repos\\espfw\\controller-firmware\\build\\CMakeCache.txt","stack":"Error: ENOENT: no such file or directory, access 'c:\\repos\\espfw\\controller-firmware\\build\\CMakeCache.txt'\n\tat Object.access (node:internal/fs/sync:42:11)\n\tat Object.accessSync (node:fs:248:10)\n\tat t.accessSync (node:electron/js2c/node_init:2:8888)\n\tat Se (\\dist\\extension.js:2:1461757)\n\tat \\dist\\extension.js:2:1855029","level":"error","timestamp":"2024-07-23T14:57:39.899Z"}
{"user":true,"level":"warn","message":"There is no CMakeCache.txt. Please try to delete the build directory manually.","timestamp":"2024-07-23T14:57:39.900Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:163:20722)\n\tat l.y (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:82:664)\n\tat l.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:82:881)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:156:57296)","level":"error","timestamp":"2024-07-23T14:57:44.261Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:163:20722)\n\tat l.y (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:82:664)\n\tat l.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:82:881)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:156:57296)","level":"error","timestamp":"2024-07-23T14:58:08.048Z"}
{"message":"Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n","stack":"Error: Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n\n\tat \\dist\\extension.js:2:1464939\n\tat ChildProcess.exithandler (node:child_process:415:7)\n\tat ChildProcess.emit (node:events:514:28)\n\tat maybeClose (node:internal/child_process:1105:16)\n\tat Socket.<anonymous> (node:internal/child_process:457:11)\n\tat Socket.emit (node:events:514:28)\n\tat Pipe.<anonymous> (node:net:337:12)","level":"error","timestamp":"2024-07-23T15:00:03.630Z"}
{"message":"Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n","stack":"Error: Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n\n\tat \\dist\\extension.js:2:1464939\n\tat ChildProcess.exithandler (node:child_process:415:7)\n\tat ChildProcess.emit (node:events:514:28)\n\tat maybeClose (node:internal/child_process:1105:16)\n\tat Socket.<anonymous> (node:internal/child_process:457:11)\n\tat Socket.emit (node:events:514:28)\n\tat Pipe.<anonymous> (node:net:337:12)","level":"error","timestamp":"2024-07-23T15:00:04.995Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:164:20994)\n\tat r.B (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:737)\n\tat r.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:954)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:157:57378)","level":"error","timestamp":"2024-07-23T15:02:53.210Z"}
{"message":"Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n","stack":"Error: Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n\n\tat \\dist\\extension.js:2:1464939\n\tat ChildProcess.exithandler (node:child_process:415:7)\n\tat ChildProcess.emit (node:events:514:28)\n\tat maybeClose (node:internal/child_process:1105:16)\n\tat Socket.<anonymous> (node:internal/child_process:457:11)\n\tat Socket.emit (node:events:514:28)\n\tat Pipe.<anonymous> (node:net:337:12)","level":"error","timestamp":"2024-07-23T15:05:18.528Z"}
{"message":"Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n","stack":"Error: Open On-Chip Debugger  v0.11.0-esp32-20220706 (2022-07-06-15:48)\r\nLicensed under GNU GPL v2\r\nFor bug reports, read\r\n\thttp://openocd.org/doc/doxygen/bugs.html\r\n\n\tat \\dist\\extension.js:2:1464939\n\tat ChildProcess.exithandler (node:child_process:415:7)\n\tat ChildProcess.emit (node:events:514:28)\n\tat maybeClose (node:internal/child_process:1105:16)\n\tat Socket.<anonymous> (node:internal/child_process:457:11)\n\tat Socket.emit (node:events:514:28)\n\tat Pipe.<anonymous> (node:net:337:12)","level":"error","timestamp":"2024-07-23T15:05:20.496Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:164:20994)\n\tat r.B (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:737)\n\tat r.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:954)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:157:57378)","level":"error","timestamp":"2024-07-23T15:06:08.772Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:164:20994)\n\tat r.B (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:737)\n\tat r.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:954)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:157:57378)","level":"error","timestamp":"2024-07-23T15:06:27.873Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:164:20994)\n\tat r.B (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:737)\n\tat r.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:954)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:157:57378)","level":"error","timestamp":"2024-07-23T15:06:39.527Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:164:20994)\n\tat r.B (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:737)\n\tat r.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:954)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:157:57378)","level":"error","timestamp":"2024-07-23T15:11:50.695Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:164:20994)\n\tat r.B (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:737)\n\tat r.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:954)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:157:57378)","level":"error","timestamp":"2024-07-23T15:12:11.452Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:164:20994)\n\tat r.B (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:737)\n\tat r.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:954)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:157:57378)","level":"error","timestamp":"2024-07-23T15:12:13.585Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:164:20994)\n\tat r.B (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:737)\n\tat r.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:954)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:157:57378)","level":"error","timestamp":"2024-07-23T15:52:55.855Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: Task ESP-IDF Compile exited with code 1\n\tat \\dist\\extension.js:2:1590016\n\tat n.value (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:164:20994)\n\tat r.B (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:737)\n\tat r.fire (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:83:954)\n\tat n.$onDidEndTaskProcess (\\out\\vs\\workbench\\api\\node\\extensionHostProcess.js:157:57378)","level":"error","timestamp":"2024-07-23T15:53:07.471Z"}
{"message":"Cannot access filePath: ${env:ADF_PATH}","stack":"Error: ENOENT: no such file or directory, access 'C:\\Users\\RickyTheil\\AppData\\Local\\Programs\\Microsoft VS Code\\${env:ADF_PATH}'\n\tat Object.access (node:internal/fs/sync:42:11)\n\tat Object.accessSync (node:fs:248:10)\n\tat t.accessSync (node:electron/js2c/node_init:2:8888)\n\tat Se (\\dist\\extension.js:2:1461757)\n\tat \\dist\\extension.js:2:1685466\n\tat Fs (\\dist\\extension.js:2:1686534)\n\tat async \\dist\\extension.js:2:1902761","level":"error","timestamp":"2024-07-23T15:59:56.456Z"}
{"message":"Cannot access filePath: ${env:MDF_PATH}","stack":"Error: ENOENT: no such file or directory, access 'C:\\Users\\RickyTheil\\AppData\\Local\\Programs\\Microsoft VS Code\\${env:MDF_PATH}'\n\tat Object.access (node:internal/fs/sync:42:11)\n\tat Object.accessSync (node:fs:248:10)\n\tat t.accessSync (node:electron/js2c/node_init:2:8888)\n\tat Se (\\dist\\extension.js:2:1461757)\n\tat \\dist\\extension.js:2:1685644\n\tat Fs (\\dist\\extension.js:2:1686534)\n\tat async \\dist\\extension.js:2:1902761","level":"error","timestamp":"2024-07-23T15:59:56.457Z"}
{"message":"Cannot access filePath: ${env:ESP_MATTER_PATH}","stack":"Error: ENOENT: no such file or directory, access 'C:\\Users\\RickyTheil\\AppData\\Local\\Programs\\Microsoft VS Code\\${env:ESP_MATTER_PATH}'\n\tat Object.access (node:internal/fs/sync:42:11)\n\tat Object.accessSync (node:fs:248:10)\n\tat t.accessSync (node:electron/js2c/node_init:2:8888)\n\tat Se (\\dist\\extension.js:2:1461757)\n\tat \\dist\\extension.js:2:1685736\n\tat Fs (\\dist\\extension.js:2:1686534)\n\tat async \\dist\\extension.js:2:1902761","level":"error","timestamp":"2024-07-23T15:59:56.457Z"}
{"message":"Cannot access filePath: ${env:HOMEKIT_PATH}","stack":"Error: ENOENT: no such file or directory, access 'C:\\Users\\RickyTheil\\AppData\\Local\\Programs\\Microsoft VS Code\\${env:HOMEKIT_PATH}'\n\tat Object.access (node:internal/fs/sync:42:11)\n\tat Object.accessSync (node:fs:248:10)\n\tat t.accessSync (node:electron/js2c/node_init:2:8888)\n\tat Se (\\dist\\extension.js:2:1461757)\n\tat \\dist\\extension.js:2:1685832\n\tat Fs (\\dist\\extension.js:2:1686534)\n\tat async \\dist\\extension.js:2:1902761","level":"error","timestamp":"2024-07-23T15:59:56.457Z"}
{"message":"spawn \\Git\\cmd\\git.exe ENOENT","stack":"Error: spawn \\Git\\cmd\\git.exe ENOENT\n\tat ChildProcess._handle.onexit (node:internal/child_process:286:19)\n\tat onErrorNT (node:internal/child_process:484:16)\n\tat process.processTicksAndRejections (node:internal/process/task_queues:82:21)","level":"error","timestamp":"2024-07-23T15:59:56.459Z"}
{"user":true,"level":"warn","message":"Extension configuration report has been copied to clipboard with errors","timestamp":"2024-07-23T15:59:56.459Z"}

--------------------------------------------------------------------------------------------------------------------------------------------


### Extension

_No response_

### Description

Can no longer build from scratch after build directory is deleted. Error is "CMake Error: Could not create named generator =Ninja". This is when using the Build option from the vs code command palette.

This is because the plugin is now adding a = sign between the -G option and the "Ninja" value
![image](https://github.com/user-attachments/assets/7429c232-3eea-45e1-8959-37b446060557)

If I go into an ESP-IDF terminal and run idf.py build, it will build and then I can use the command palette to build. This problem only started when the extension was updated to 1.8. I purposefully no longer update this tool because these kinds of minor problems are all too common, but in this case vs code auto updated it. What's strange is that I downgraded the extension to 1.7.x and the problem still exists.

Even if I try to run that command in an idf terminal and fix the -G option, it still doesn't work
![image](https://github.com/user-attachments/assets/2f89c2de-fdf9-45d4-8712-c23ba320db8b)


This is because the -G, -B and -S options don't use the = sign and never have
![image](https://github.com/user-attachments/assets/233a2c02-1af9-43e3-af15-d1f436afaa8c)


### Debug Message

```plain
N/A
```


### Other Steps to Reproduce

_No response_

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

--- a/package.json
+++ b/package.json
@@ -575,11 +575,7 @@
           },
           "idf.cmakeCompilerArgs": {
             "type": "array",
-            "default": [
-              "-G=Ninja",
-              "-DPYTHON_DEPS_CHECKED=1",
-              "-DESP_PLATFORM=1"
-            ],
+            "default": [],
             "description": "%param.cmakeCompilerArgs%",
             "scope": "resource"
           },

--- a/src/build/buildTask.ts
+++ b/src/build/buildTask.ts
@@ -152,11 +152,14 @@ export class BuildTask {
           "-DESP_PLATFORM=1",
         ];
       }
-      let compilerArgs =
-        (idfConf.readParameter(
-          "idf.cmakeCompilerArgs",
-          this.currentWorkspace
-        ) as Array<string>) || defaultCompilerArgs;
+      let compilerArgs = idfConf.readParameter(
+        "idf.cmakeCompilerArgs",
+        this.currentWorkspace
+      ) as Array<string>;
+
+      if (!compilerArgs || compilerArgs.length === 0) {
+        compilerArgs = defaultCompilerArgs;
+      }
       let buildPathArgsIndex = compilerArgs.indexOf("-B");
       if (buildPathArgsIndex !== -1) {
         compilerArgs.splice(buildPathArgsIndex, useEqualSign ? 1 : 2);

--- a/src/build/buildTask.ts
+++ b/src/build/buildTask.ts
@@ -128,6 +128,7 @@ export class BuildTask {
     if (!cmakeCacheExists) {
       const espIdfVersion = await getEspIdfFromCMake(this.idfPathDir);
       let defaultCompilerArgs;
+      const useEqualSign = compareVersion(espIdfVersion, "4.4") >= 0;
       if (espIdfVersion === "x.x") {
         Logger.warn(
           "Could not determine ESP-IDF version. Using default compiler arguments for the latest known version."
@@ -137,7 +138,7 @@ export class BuildTask {
           "-DPYTHON_DEPS_CHECKED=1",
           "-DESP_PLATFORM=1",
         ];
-      } else if (compareVersion(espIdfVersion, "4.4") >= 0) {
+      } else if (useEqualSign) {
         defaultCompilerArgs = [
           "-G=Ninja",
           "-DPYTHON_DEPS_CHECKED=1",
@@ -158,12 +159,19 @@ export class BuildTask {
         ) as Array<string>) || defaultCompilerArgs;
       let buildPathArgsIndex = compilerArgs.indexOf("-B");
       if (buildPathArgsIndex !== -1) {
-        compilerArgs.splice(buildPathArgsIndex, 2);
+        compilerArgs.splice(buildPathArgsIndex, useEqualSign ? 1 : 2);
       }
-      compilerArgs.push(`-B=${this.buildDirPath}`);
-
+      if (useEqualSign) {
+        compilerArgs.push(`-B=${this.buildDirPath}`);
+      } else {
+        compilerArgs.push("-B", this.buildDirPath);
+      }
       if (compilerArgs.indexOf("-S") === -1) {
-        compilerArgs.push(`-S=${this.currentWorkspace.fsPath}`);
+        if (useEqualSign) {
+          compilerArgs.push(`-S=${this.currentWorkspace.fsPath}`);
+        } else {
+          compilerArgs.push("-S", this.currentWorkspace.fsPath);
+        }
       }
 
       const sdkconfigDefaults =

--- a/src/build/buildTask.ts
+++ b/src/build/buildTask.ts
@@ -21,7 +21,12 @@ import { join } from "path";
 import { Logger } from "../logger/logger";
 import * as vscode from "vscode";
 import * as idfConf from "../idfConfiguration";
-import { appendIdfAndToolsToPath, isBinInPath } from "../utils";
+import {
+  appendIdfAndToolsToPath,
+  isBinInPath,
+  getEspIdfFromCMake,
+  compareVersion,
+} from "../utils";
 import { TaskManager } from "../taskManager";
 import { selectedDFUAdapterId } from "../flash/dfu";
 
@@ -121,14 +126,36 @@ export class BuildTask {
         : vscode.TaskRevealKind.Silent;
 
     if (!cmakeCacheExists) {
-      let compilerArgs = (idfConf.readParameter(
-        "idf.cmakeCompilerArgs",
-        this.currentWorkspace
-      ) as Array<string>) || [
-        "-G=Ninja",
-        "-DPYTHON_DEPS_CHECKED=1",
-        "-DESP_PLATFORM=1",
-      ];
+      const espIdfVersion = await getEspIdfFromCMake(this.idfPathDir);
+      let defaultCompilerArgs;
+      if (espIdfVersion === "x.x") {
+        Logger.warn(
+          "Could not determine ESP-IDF version. Using default compiler arguments for the latest known version."
+        );
+        defaultCompilerArgs = [
+          "-G=Ninja",
+          "-DPYTHON_DEPS_CHECKED=1",
+          "-DESP_PLATFORM=1",
+        ];
+      } else if (compareVersion(espIdfVersion, "4.4") >= 0) {
+        defaultCompilerArgs = [
+          "-G=Ninja",
+          "-DPYTHON_DEPS_CHECKED=1",
+          "-DESP_PLATFORM=1",
+        ];
+      } else {
+        defaultCompilerArgs = [
+          "-G",
+          "Ninja",
+          "-DPYTHON_DEPS_CHECKED=1",
+          "-DESP_PLATFORM=1",
+        ];
+      }
+      let compilerArgs =
+        (idfConf.readParameter(
+          "idf.cmakeCompilerArgs",
+          this.currentWorkspace
+        ) as Array<string>) || defaultCompilerArgs;
       let buildPathArgsIndex = compilerArgs.indexOf("-B");
       if (buildPathArgsIndex !== -1) {
         compilerArgs.splice(buildPathArgsIndex, 2);

